account_banking_ach_discount/models/account_move.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (C) 2019 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def _get_reconciled_info_JSON_values(self):
+        res = super(AccountMove, self)._get_reconciled_info_JSON_values()
+        inv_number = self.ref
+        if res:
+            flag = False
+            for item in res:
+                payment_lines = set()
+                for line in self.line_ids:
+                    payment_lines.update(
+                        line.mapped("matched_credit_ids.credit_move_id.id")
+                    )
+                    payment_lines.update(
+                        line.mapped("matched_debit_ids.debit_move_id.id")
+                    )
+                    payment_move_line_ids = (
+                        self.env["account.move.line"]
+                        .browse(list(payment_lines))
+                        .sorted()
+                    )
+
+                for mvl in payment_move_line_ids:
+                    # get bank payment line
+                    if mvl.move_id and mvl.id == item["payment_id"]:
+                        for pay_li in mvl.bank_payment_line_id.payment_line_ids:
+                            # Get related payment line ref
+                            if pay_li.communication == inv_number:
+                                item["amount"] = pay_li.amount_currency
+                    # for non-ach payment
+                    # Deduct the discount only for the related payment.
+                    # Discount is applied on the last payment (i.e. fully reconciled).
+                    if (
+                        not mvl.bank_payment_line_id
+                        and mvl.move_id == self.id
```

Comment:
please change it to mvl.move_id.id == self.id

Reply:
Even better, change to `mvl.move_id == self`